corehq/apps/hqwebapp/views.py

```diff
@@ -327,6 +330,12 @@ def csrf_failure(request, reason=None, template_name="csrf_failure.html"):
 @sensitive_post_parameters('auth-password')
 def _login(req, domain_name, template_name):
 
+    if 'auth-username' in req.POST:
+        couch_user = CouchUser.get_by_username(req.POST['auth-username'])
```

Comment:
It's still a bit confusing to me that `req.couch_user` doesn't exist. It should get set here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/middleware.py#L47 and (I think) generally be able to be set if `req.user` is set (which is used in the next `if` block)

Also looking down at line 350 where `req.POST['auth-username']` is occasionally formatted. Does that mean that this call should go after that `if` block?

Reply:
I stuck a breakpoint in there and saw that `req.couch_user` doesn't get added for the arbitrary user I was signing in as because of this line: `if request.user.is_anonymous and 'domain' in view_kwargs:`, `view_kwargs` doesn't have a `domain` key. 
That's the same reason it doesn't get added here either: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/middleware.py#L60

From playing with this, found out that that means: the `domain` key (and therefore `req.couch_user`) exists if you click on a link to a page in a specific domain, but if you go straight to the login at commcarehq.org, it doesn't exist.

Reply:
And regarding the formatting, that checks for an `@`, and if it doesn't find one, it changes `<username>` to `<username>@<domain>.commcarehq.org`
That's a mobile worker thing, right?

Reply:
Cool, moved the validation here: http://localhost:8000/a/xv/settings/users/commcare/
(and made sure the language gets changed for mobile workers logging into hq with a different language)